src/controllers/authenticationController.ts

```diff
@@ -517,6 +550,80 @@ export default class AuthenticationController {
         );
       });
 
+    if (
+      userEmail &&
+      this.lollipopParams.isLollipopEnabled &&
+      isUserElegibleForFastLoginResult
+    ) {
+      const errorOrNotifyLoginResult = await pipe(
+        {
+          email: userEmail,
+          family_name: user.family_name,
+          fiscal_code: user.fiscal_code,
+          identity_provider: spidUser.issuer ?? "cie",
+          ip_address: pipe(
+            errorOrUserIp,
+            // we've already checked errorOrUserIp, this will never happen
+            E.getOrElse(() => "")
+          ),
+          name: user.name,
+        },
+        UserLoginParams.decode,
+        TE.fromEither,
+        TE.mapLeft((err) => {
+          {
+            log.error(
+              "Cannot decode UserLoginParams",
+              readableReportSimplified(err)
+            );
+            return readableReportSimplified(err);
+          }
+        }),
+        TE.chainW(
+          flow(
+            (loginData) => this.onUserLogin(loginData),
+            TE.mapLeft((err) => {
+              {
+                log.error(`Unable to notify user login event | ${err.message}`);
+                return err.message;
+              }
+            })
+          )
+        ),
+        TE.mapLeft((err) => {
+          this.appInsightsTelemetryClient?.trackEvent({
+            name: lollipopErrorEventName + ".notify",
+            properties: {
+              error: err,
+              fiscal_code: sha256(user.fiscal_code),
+              message: "acs: Unable to notify user login event",
+            },
+          });
+
+          return ResponseErrorInternal("Unable to notify user login event");
+        })
+      )();
+
+      if (E.isLeft(errorOrNotifyLoginResult)) {
+        const delLollipopResult =
+          await this.sessionStorage.delLollipopDataForUser(user.fiscal_code);
```

Comment:
What if we send the email before setting all the lollipop and session data?

Reply:
It requires some effort, we can refactor it in another PR